src/_pytest/mark/legacy.py

```diff
@@ -46,7 +46,7 @@ def from_item(cls, item: "Item") -> "KeywordMapping":
         import pytest
 
         for item in item.listchain():
-            if not isinstance(item, pytest.Instance):
+            if item.parent and not isinstance(item, pytest.Instance):
```

Comment:
Is `item.parent` the canonical way to check for the root node? And are we sure we always want to exclude the root node from keyword matching?

In any case a comment explaining the `item.parent` condition would be helpful I think since it's not immediately obvious.

Reply:
I think the previous fix is better, because it explicitly skips `Session` in the code. 👍